tests/phpunit/agentic-commerce/class-wc-stripe-agentic-commerce-inventory-tracker-test.php

```
@@ -677,6 +729,59 @@ public function test_track_product_archive_stores_pending_archive() {
+	 * @return void
+	 */
+	public function test_track_product_archive_evicts_stale_entries_when_filter_now_excludes() {
+		$product = $this->create_simple_product_with_stock( 7 );
+
+		// Seed both queues while the filter still allows it.
+		$this->sut->track_stock_change( $product );
+		$this->sut->track_product_archive( $product );
+		$this->assertArrayHasKey( $product->get_id(), get_option( WC_Stripe_Agentic_Commerce_Inventory_Tracker::PENDING_ARCHIVES_OPTION, [] ) );
+
+		// Flip the filter, then trigger another archive event.
+		$callback = static fn( $sync, $candidate ) => $candidate->get_id() !== $product->get_id();
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$sync'. (undefined)

(UnusedFormalParameter)